src/lazy/encoder/writer.rs

```diff
@@ -846,4 +894,86 @@ mod tests {
             ],
         )
     }
+
+    /// Writes a struct with all of the provided field names using the requested field name encoding.
+    /// For simplicity, the value for each field is the integer 0.
+    fn struct_field_encoding_test(
+        encoding: FieldNameEncoding,
+        field_names_and_encodings: &[(RawSymbolRef, &[u8])],
+    ) -> IonResult<()> {
+        // Configure a struct writer that uses the requested field name encoding
+        let mut writer = Writer::new(v1_1::Binary, Vec::new())?;
+        let mut struct_writer = writer
+            .value_writer()
+            .struct_writer()?
+            .with_field_name_encoding(encoding);
+
+        for (name, _) in field_names_and_encodings {
+            struct_writer.write(name, /* same value for every field*/ 0)?;
+        }
+        struct_writer.close()?;
+        let bytes = writer.close()?;
+
+        let mut reader = SystemReader::new(v1_1::Binary, bytes.as_slice());
+        let struct_ = reader.expect_next_value()?.read()?.expect_struct()?;
+        for (field, (_name, expected_encoding)) in
+            struct_.iter().zip(field_names_and_encodings.iter())
+        {
+            let raw_name = field?.get_raw_name().unwrap();
+            let raw_name_encoding = raw_name.span().bytes();
+            assert_eq!(
+                raw_name_encoding, *expected_encoding,
+                "actual {:02X?}\n!=\nexpected {:02X?}",
+                raw_name_encoding, *expected_encoding
+            );
+        }
+
+        Ok(())
+    }
+
+    #[test]
+    fn intern_all_field_names() -> IonResult<()> {
+        struct_field_encoding_test(
+            FieldNameEncoding::WriteAsSymbolIds,
+            &[
+                // New symbols
+                (RawSymbolRef::Text("foo"), &[0x15]), // FlexUInt SID $10,
+                (RawSymbolRef::Text("bar"), &[0x17]), // FlexUInt SID $11,
+                (RawSymbolRef::Text("baz"), &[0x19]), // FlexUInt SID $12,
+                // Symbols that are already in the symbol table
+                (RawSymbolRef::Text("name"), &[0x09]), // FlexUInt SID $4,
+                (RawSymbolRef::Text("foo"), &[0x15]),  // FlexUInt SID $10,
+            ],
+        )
+    }
+
+    #[test]
+    fn write_all_field_names_as_text() -> IonResult<()> {
+        struct_field_encoding_test(
+            FieldNameEncoding::WriteAsInlineText,
+            &[
+                // New symbols
+                (RawSymbolRef::Text("foo"), &[0xFB, 0x66, 0x6F, 0x6F]), // FlexSym -3, "foo"
+                (RawSymbolRef::Text("bar"), &[0xFB, 0x62, 0x61, 0x72]), // FlexSym -3, "bar"
+                (RawSymbolRef::Text("baz"), &[0xFB, 0x62, 0x61, 0x7A]), // FlexSym -3, "baz"
+                // Symbols that are already in the symbol table are written as SIDs
+                (RawSymbolRef::Text("name"), &[0xF9, 0x6E, 0x61, 0x6D, 0x65]), // FlexSym 4, SID $4,
```

Comment:
This looks like a copy-paste mistake from one of the other tests. Is it?

Reply:
Yeah. The close is right but the comments are wrong. I'll fix this before merging, thanks.